### PR TITLE
#155059934 - Add logic for incident timeline

### PR DESCRIPTION
**Pull request by NaiRobley**
#### What does this PR do?
Implement Incident Timeline Logic

#### Description of Task to be completed?
As an Admin, WHEN I click on an incident subject, THEN I should be able to:-
- see an incident's details and activity on the sidebar
- see the chat history with the reporter and be able to post messages
- see the notes associated with the incident and be able to add new notes, edit or archive existing notes.

#### Where should the reviewer start?
- View file changes on PR

#### How should this be manually tested?
- Create a .env file based on .env.template.
- Run in development mode by setting NODE_ENV=development in .env
- Open the dashboard and click on an incident's subject to view the timeline page

#### What are the relevant pivotal tracker stories?
[#155059934](https://www.pivotaltracker.com/story/show/155059934)

#### Screenshots (if appropriate)
With chats selected:
image

With notes selected:
image

### CODE REVIEW

File: __tests__/CustomMenu.test.jsx [hunk 1]
```diff
@@ -4,10 +4,9 @@ import { shallow } from 'enzyme';
...
-        const customMenu = shallow(<CustomMenu />);
-        const tree = shallowToJSON(customMenu);
-        expect(tree).toMatchSnapshot();
-    });
-
+  it('Should have all the CustomMenu content', () => {
+    const customMenu = shallow(<CustomMenu />);
+    const tree = shallowToJSON(customMenu);
+    expect(tree).toMatchSnapshot();
+  });
```
[the remark this answers is not in the document]
Comment (NaiRobley) [reply __tests__/CustomMenu.test.jsx]: This was the initial testing done by @RayNjeri

**Comment by kimobrian** [comment]
image

@NaiRobley work on the mockup API to ensure that we can create a note and a chat locally. It creates but there's a lot of console errors.

File: src/mock_endpoints/mockControllers.js [hunk 2]
```diff
@@ -68,19 +71,75 @@ module.exports = {
     return incidentChats;
   },
   changeStatus: (incidentId, statusId) => {
-    return this.getIncident(incidentId);
+    let incident = incidents.find(incident => {
+      return incident.id == incidentId;
```
Comment (kimobrian) [inline src/mock_endpoints/mockControllers.js]: Let's maintain `===` instead of `==`

File: src/mock_endpoints/mockControllers.js [hunk 3]
```diff
@@ -68,19 +71,75 @@ module.exports = {
...
+      return location.id === incident.locationId;
+    });
+    return incident;
   },
   changeAssignee: (incidentId, assigneeId) => {
-    return this.getIncident(incidentId);
+    let incident = incidents.find(incident => {
+      return incident.id == incidentId;
+    });
+    let newAssignee = users.find(user => {
```
Comment (kimobrian) [inline src/mock_endpoints/mockControllers.js]: Is this being used?

File: src/mock_endpoints/mockControllers.js [hunk 4]
```diff
@@ -68,19 +71,75 @@ module.exports = {
...
+      }) || null;
+    incident['Location'] = locations.find(location => {
+      return location.id === incident.locationId;
+    });
+    return incident;
   },
   addNote: (incidentId, userId, note) => {
     let newNote = {
-      id: chats.length++,
+      id: notes.length++,
```
Comment (kimobrian) [inline src/mock_endpoints/mockControllers.js]: Let's not do this her. Should be handled by the API

File: src/mock_endpoints/mockControllers.js (hunk 3 above)
Comment (NaiRobley) [reply src/mock_endpoints/mockControllers.js]: No. I have removed it

**Comment by NaiRobley** [comment]
@kimobrian fixed the console and lint errors, and maintained `===` instead of `==`. Also removed the unused variables

**Pull request merged**